### PR TITLE
Feedback & fixes

### PR DESCRIPTION
**Pull request by elijahr**
- Fix issue where unauthenticated `hub` could block CI waiting for user input
- Search for build-nim binaries via local text file cache, fall back to using `hub` only if auth credentials are present
- Fix issue on aarch64/arm64 where OS var hadn't been set yet
- Feedback from @smorimoto via https://github.com/asdf-vm/asdf-plugins/pull/343:
  - Test plugin against latest nim nightly
  - Pin plugin actions to v1

### CODE REVIEW

File: .github/workflows/build-latest-nightly.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          # - os: macos-latest
+          #   nim-version: latest
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Test nimble ${{ matrix.nim-version }}
+        uses: asdf-vm/actions/plugin-test@v1
+        with:
+          command: nim -v
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (smorimoto) [inline .github/workflows/build-latest-nightly.yml:27]: You can remove this line.

Comment (elijahr) [reply .github/workflows/build-latest-nightly.yml]: Removed in 1a414c1d87d9fed2128ad1fd9ccee370c1d0c91f

Comment (smorimoto) [reply .github/workflows/build-latest-nightly.yml]: It’s same to `.github/workflows/build.yml`

Comment (elijahr) [reply .github/workflows/build-latest-nightly.yml]: Removed in 1e9e7ed3819a4ed39aa5fae3595b91a6b9d8767f

Comment (smorimoto) [reply .github/workflows/build-latest-nightly.yml]: Looks good!

**Review by smorimoto — APPROVED** [review]

**Pull request merged**